### PR TITLE
APPDEV-1199 Two times making getUser request call on pin code validation screen

### PR DESCRIPTION
**Pull request by vishal-revdiwala**
- Refactored code to avoid extra pin validation call

### CODE REVIEW

File: Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift [hunk 1]
```diff
@@ -242,6 +186,22 @@ extension LoginViewController: CodeInputViewDelegate {
         })
     }
     
+    fileprivate func allowedNumberOfLoginAttemptsFailed() {
```
Comment (Bert-R) [inline Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift]: Functions normally are named after what they do (e.g. ``registerFailedLoginAttempt``), with event handlers as the exception (e.g. ``onTooManyLoginAttempts``). In this particular case, the method name doesn't explain me the function of the function (pun intended).

Is this something like ``showMaxLoginAttemptsReachedScreen``?

File: Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift [hunk 2]
```diff
@@ -9,15 +9,19 @@
...
-    fileprivate var totalAttempts : Int = 3
+class LoginViewController: LoginSignupValidationMasterView {
+    
     @IBOutlet weak var bottomSpaceContraint: NSLayoutConstraint!
     @IBOutlet var pinResetButton: UIButton!
     @IBOutlet var closeButton: UIBarButtonItem?
     @IBOutlet var accountBlockedTitle: UILabel?
+    
+    var loginAttempts:Int = 1
+    var totalAttempts : Int = 3
```
Comment (Bert-R) [inline Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift]: This is the maximum, right? Then name like that: ``maxAttempts``.

File: Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift [hunk 3]
```diff
@@ -9,15 +9,19 @@
...
-    var loginAttempts:Int = 1
-    fileprivate var totalAttempts : Int = 3
+class LoginViewController: LoginSignupValidationMasterView {
+    
     @IBOutlet weak var bottomSpaceContraint: NSLayoutConstraint!
     @IBOutlet var pinResetButton: UIButton!
     @IBOutlet var closeButton: UIBarButtonItem?
     @IBOutlet var accountBlockedTitle: UILabel?
+    
+    var loginAttempts:Int = 1
```
Comment (Bert-R) [inline Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift]: Does this number survive an app restart? Or can the user do an attempt, close the app and do another, for as many times as they want?

Comment (vishal-revdiwala) [reply Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift]: Code written such a way that **loginAttempts** number won't survive if user restart the app.

Comment (Bert-R) [reply Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift]: Then please create a ticket to correct that in a next release.  Talk to @Sivateja-CMI too, to see whether the ticket should be cloned for Android.

Comment (vishal-revdiwala) [reply Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift]: Ok sure. Is this need to updated to wiki or it's already there?

File: Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift (hunk 2 above)
Comment (vishal-revdiwala) [reply Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift]: Yes. Updated.

File: Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift (hunk 1 above)
Comment (vishal-revdiwala) [reply Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift]: Yes, you are right. I've renamed to `showMaxLoginAttemptsReachedScreen`.

**Comment by vishal-revdiwala** [comment]
Here [APPDEV-1200](https://jira.yona.nu/browse/APPDEV-1200) for the persistent storage of the number of login attempts.

**Pull request merged**